apps/web/src/pages/integrations/components/multi-provider/v2/UpdateProviderSidebarV2.tsx

```diff
@@ -0,0 +1,424 @@
+import styled from '@emotion/styled';
+import { Box, Center, Group } from '@mantine/core';
+import { useClipboard, useDisclosure } from '@mantine/hooks';
+import { Button, Check, colors, Copy, Input, Sidebar, Text } from '@novu/design-system';
+import {
+  CHANNELS_WITH_PRIMARY,
+  CredentialsKeyEnum,
+  EmailProviderIdEnum,
+  IConfigCredentials,
+  IConstructIntegrationDto,
+  ICredentialsDto,
+  InAppProviderIdEnum,
+  SmsProviderIdEnum,
+} from '@novu/shared';
+import { useEffect, useMemo, useState } from 'react';
+import { Controller, FormProvider, useForm, useWatch } from 'react-hook-form';
+import slugify from 'slugify';
+import { useWebhookSupportStatus } from '../../../../../api/hooks';
+import { useUpdateIntegration } from '../../../../../api/hooks/useUpdateIntegration';
+import { Conditions, IConditions } from '../../../../../components/conditions';
+import { When } from '../../../../../components/utils/When';
+import { useEnvironment } from '../../../../../hooks';
+import { successMessage } from '../../../../../utils/notifications';
+import { Faq } from '../../../../quick-start/components/QuickStartWrapper';
+import { SetupTimeline } from '../../../../quick-start/components/SetupTimeline';
+import { FrameworkEnum } from '../../../../quick-start/consts';
+import { defaultIntegrationConditionsProps } from '../../../constants';
+import type { IIntegratedProvider } from '../../../types';
+import { useProviders } from '../../../useProviders';
+import { IntegrationInput } from '../../IntegrationInput';
+import { ShareableUrl } from '../../Modal/ConnectIntegrationForm';
+import { NovuInAppFrameworkHeader } from '../../NovuInAppFrameworkHeader';
+import { SetupWarning } from '../../SetupWarning';
+import { UpdateIntegrationCommonFields } from '../../UpdateIntegrationCommonFields';
+import { UpdateIntegrationSidebarHeader } from '../../UpdateIntegrationSidebarHeader';
+import { NovuInAppFrameworks } from '../../v2';
+import { NovuProviderSidebarContent } from '../NovuProviderSidebarContent';
+import { useSelectPrimaryIntegrationModal } from '../useSelectPrimaryIntegrationModal';
+
+interface IProviderForm {
+  name: string;
+  credentials: ICredentialsDto;
+  active: boolean;
+  identifier: string;
+  conditions: IConditions[];
+}
+
+enum SidebarStateEnum {
+  NORMAL = 'normal',
+  EXPANDED = 'expanded',
+}
+
+export function UpdateProviderSidebar({
```

Comment:
☑ **chore:** I'm not sure if that is a good idea to duplicate this component. Wouldn't that be easier just to update it internally based on ff? The biggest part of the changes is the frameworks list.
wdyt?

Reply:
Actually, I noticed it is a pattern the team is following to create a new component for v2.
As the whole onboarding flow for the `inbox` will also get updated in a separate ticket by the team workflow.
I figured it will be easier to add a new v2 component which the team will use to add the new onboarding flow.

Reply:
Thanks for the clarification, if that's the pattern, let's follow it 🙌